File: server/models/expense.js
const mongoose = require('mongoose');

const expenseSchema = mongoose.Schema({

    name: {
        type: String,
        required: true,
        trim: true
    },

    description: {
        type: String,
        required: true,
        trim: true
    },

    amount: {
        type: String,
        required: true
    },
   
    date: {
        type: Date,
        required: true,
    },

    category: {
        type: String,
        require: true
    },

    expenserId: {
        type: String,
    },

    expenserName: {
        type: String,
    },

    

});

const Expense = mongoose.model('Expense', expenseSchema);
module.exports = { Expense, expenseSchema };

File: lib/features/expenses/domain/entity/expense.dart
// ignore_for_file: public_member_api_docs, sort_constructors_first
import 'dart:convert';

class Expense {
  final String? id;
  final String expenserId;
  final String expenserName;
  final String name;
  final String description;
  final String amount;
  final DateTime date;
  final String category;
  final bool isEdited;

  Expense({
    this.id,
    required this.expenserId,
    required this.expenserName,
    required this.name,
    required this.description,
    required this.amount,
    required this.date,
    required this.category,
    required this.isEdited,
  });
}
